### PR TITLE
`azurerm_data_factory_dataset_{cosmosdb_sqlapi|mysql|postgresql}` - Fix incorrect type/error message during read

### PR DESCRIPTION
**Pull request by magodo**
`azurerm_data_factory_dataset_{cosmosdb_sqlapi|mysql|postgresql}` - Fix incorrect type/error message during read

### CODE REVIEW

File: internal/services/datafactory/data_factory_dataset_postgresql_resource.go [hunk 1]
```diff
@@ -249,7 +249,7 @@ func resourceDataFactoryDatasetPostgreSQLRead(d *pluginsdk.ResourceData, meta in
 	d.Set("name", id.Name)
 	d.Set("data_factory_id", dataFactoryId.ID())
 
-	postgresqlTable, ok := resp.Properties.AsRelationalTableDataset()
+	postgresqlTable, ok := resp.Properties.AsPostgreSQLTableDataset()
```
Comment (mbfrahry) [inline internal/services/datafactory/data_factory_dataset_postgresql_resource.go:252]: This breaks the resource
```
Error: classifying Data Factory Dataset PostgreSQL Data Set: (Name "acctestds230710174416108434" / Factory Name "acctestdf230710174416108434" / Resource Group "acctestRG-df-230710174416108434"): Expected: "PostgreSqlTable" Received: "Microsoft.DataFactory/factories/datasets"
```

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @magodo, this is looking in the right direction but it also breaks one of the resources. It also looks like the resources weren't built correctly either as the mysql dataset is being built as a [relational dataset](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/datafactory/data_factory_dataset_mysql_resource.go#L173) rather than a mysql one. It looks to be the same for the [postgresql dataset] (https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/datafactory/data_factory_dataset_postgresql_resource.go#L185) as well

**Comment by magodo** [comment]
Thank you @mbfrahry for the feedback! I was mislead by https://github.com/hashicorp/terraform-provider-azurerm/blob/e9b169893b48619ebad95fa6410b99d8f8ef6f78/internal/services/datafactory/data_factory_dataset_postgresql_resource.go#L212-L216 in the create function, while I now realized this `Type` property at the top level is actually thrown away during Marshalling:

https://github.com/hashicorp/terraform-provider-azurerm/blob/e9b169893b48619ebad95fa6410b99d8f8ef6f78/vendor/github.com/Azure/azure-sdk-for-go/services/datafactory/mgmt/2018-06-01/datafactory/models.go#L83713-L83718

This brings me to wonder whether postgresql data set is actually implemented as expected, as it is now using the same type as the mysql dataset...

Regardlessly, I've also updated the error message to make it useful, otherwise, the "Expect" always print the same (useless) resource type, instead of the variant type.

**Review by mbfrahry — APPROVED** [review]
Yea...I'm not sure if the underlying model changed or the resource was written incorrectly(by me 🙈) but mysql and postgresql definitely look to be using the wrong model. 

But these error messages look better so we can move forward with this! Thanks @magodo

**Pull request merged**